docs/admin/garbage-collection.md

```diff
@@ -0,0 +1,92 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/admin/garbage-collection.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Garbage Collection
+
+Garbage collection is managed by kubelet automatically, mainly including unreferenced
+images and dead containers. kubelet applies container garbage collection every minute
+and image garbage collection every 5 minutes.
+Note that we don't recommend external garbage collection tool generally, since it could
+break the behavior of kubelet potentially if it attempts to remove all of the containers
+which acts as the tombstone kubelet relies on. Yet those garbage collector aims to deal
+with the docker leaking issues would be appreciated.
+
+### Image Collection
+
+kubernetes manages lifecycle of all images through imageManager, with the cooperation
+of cadvisor.
+The policy for garbage collecting images we apply takes two factors into consideration,
+`HighThresholdPercent` and `LowThresholdPercent`. Disk usage above the the high threshold
+will trigger garbage collection, which attempts to delete unused images until the low
+threshold is met. Least recently used images are deleted first.
+
+### Container Collection
+
+The policy for garbage collecting containers we apply takes on three variables, which can
+be user-defined. `MinAge` is the minimum age at which a container can be garbage collected,
+zero for no limit. `MaxPerPodContainer` is the max number of dead containers any single
+pod (UID, container name) pair is allowed to have, less than zero for no limit.
+`MaxContainers` is the max number of total dead containers, less than zero for no limit.
+
+kubelet sorts out containers which are unidentified or stay out of bounds set by previous
+mentioned three flags. For the latter case, the oldest containers are removed first.
+When kubelet removes the dead containers, all the files inside the container will be cleaned up as well.
+Note that we will skip the containers that are not managed by kubelet.
+
+### User Configuration
+
+Users are free to set their own value to address image garbage collection.
+
+1. `image-gc-high-threshold`, the percent of disk usage which triggers image garbage collection.
+Default is 90%.
+2. `image-gc-low-threshold`, the percent of disk usage to which image garbage collection attempts
+to free. Default is 80%.
+
+We also allow users to customize garbage collection policy, basically via following three flags.
+
+1. `minimum-container-ttl-duration`, minimum age for a finished container before it is
+garbage collected. Default is 1 minute.
+2. `maximum-dead-containers-per-container`, maximum number of old instances to retain
```

Comment:
It is probably worth mentioning that maximum-dead-containers should be large enough to allow at least 2 dead containers-per-expected-container?  May be also reference #13287 as an explanation for why these values are recommended.